meta-mender-core/recipes-mender/mender-snapshot/mender-snapshot_git.bb

```diff
@@ -0,0 +1,67 @@
+require mender-snapshot.inc
+
+# The revision listed below is not really important, it's just a way to avoid
+# network probing during parsing if we are not gonna build the git version
+# anyway. If git version is enabled, the AUTOREV will be chosen instead of the
+# SHA.
+def mender_snapshot_autorev_if_git_version(d):
+    version = d.getVar("PREFERRED_VERSION")
+    if not version:
+        version = d.getVar("PREFERRED_VERSION_%s" % d.getVar('PN'))
+    if not d.getVar("EXTERNALSRC") and version is not None and "git" in version:
+        return d.getVar("AUTOREV")
+    else:
+        return "c0c410aede66e31cd75c65e1fcb6df0b3ad91367"
```

Comment:
TODO - Needs to be updated after https://github.com/mendersoftware/mender-snapshot/pull/10 is merged

Reply:
But isn't it nice if it's something which builds? :thinking:

Reply:
The only cases where this SHA will be selected are:
* If you have no `PREFERRED_VERSION` set, in which case `DEFAULT_PREFERENCE` will prevent this recipe from being built.
* If `EXTERNALSRC` is set, in which case that source code will be used instead of whatever is in the SHA.

So it is never actually *used*. However, the Git fetcher still runs in the background, and it's both slower and "more rude" because it then connects on every build. If there is a valid SHA, however, it never hits the network.

So actually, it would be better if it didn't build, because then there is a mistake in the logic somewhere! 😄

Reply:
Well, not setting `PREFERRED_VERSION` while still not having a final tagged recipe will use this SHA, for which case is good that the recipe actually builds.

This situation for example was an issue for me when I started building `mender-flash` with yocto.

Reply:
Updated to the first building version